src/Symfony/Component/Debug/DebugClassLoader.php

```diff
@@ -313,6 +316,17 @@ public function checkAnnotations(\ReflectionClass $refl, $class)
             return $deprecations;
         }
 
+        if (
+            isset($parentAndOwnInterfaces['Serializable']) &&
+            0 === strpos($refl->getNamespaceName(), 'Symfony\\') &&
+            (
+                (!$refl->hasMethod('__serialize') || !($methodRefl = $refl->getMethod('__serialize'))->isPublic() || $methodRefl->isStatic() || !$refl->hasMethod('__unserialize') || !($methodRefl = $refl->getMethod('__unserialize'))->isPublic() || $methodRefl->isStatic()) &&
+                (!($isInterface = $refl->isInterface()) || ((self::$method[$class]['__serialize'][2] ?? true || self::$method[$class]['__unserialize'][2] ?? true)))
```

Comment:
isn't the check on `__serialize`redundant with the previous line?

Reply:
On this line it checks virtual methods of interface. By removing checks on the "static", the condition will be easier to read, ie `isset(self::$method[$class]['__serialize']`